src/config/config.rs

```diff
@@ -350,33 +347,23 @@ impl Config {
       Source::Yml => Ok(Config::from_yaml(schema)?),
     }
   }
-
   pub fn n_plus_one(&self) -> Vec<Vec<(String, String)>> {
     super::n_plus_one::n_plus_one(self)
   }
 
-  pub async fn from_file_paths(file_paths: std::slice::Iter<'_, String>) -> Result<Config> {
-    let mut config = Config::default();
-    let futures: Vec<_> = file_paths
-      .map(|file_path| async move {
-        let source = Source::detect(file_path)?;
-        let mut f = File::open(file_path).await?;
-        let mut buffer = Vec::new();
-        f.read_to_end(&mut buffer).await?;
-
-        let server_sdl = String::from_utf8(buffer)?;
-        Config::from_source(source, &server_sdl)
-      })
-      .collect();
-
-    for res in join_all(futures).await {
-      match res {
-        Ok(conf) => config = config.clone().merge_right(&conf),
-        Err(e) => return Err(e), // handle error
-      }
+  pub async fn from_file_or_url(file_paths: std::slice::Iter<'_, String>) -> Result<ConfigReader> {
+    // maintaining config reader will help to implement poll intervals in future
+    let mut config_reader = ConfigReader::init();
+    for file_path in file_paths {
+      config_reader.serialize_config(file_path).await?;
     }
+    Ok(config_reader)
+  }
 
-    Ok(config)
+  pub async fn from_file_path(file_path: &String) -> Result<Config> {
+    let source = Source::detect(file_path)?;
+    let server_sdl = ConfigReader::read_file(file_path).await;
```

Comment:
I have a little confusion about relationship between Config and ConfigReader as they both use each other. Could we consider this to make only single direction relationship? Maybe ConfigReader that creates Config with their methods and Config will contain only config entries without logic about how to parse it?

Reply:
Yes but when you need instance of config then Config::from_* makes sense however it calls ConfigReader for I/O but ConfigReader also needs instance of Config to merge them

this didn't make sense but I definitely wanted review on this 

as you suggested, I will move source detection to config reader itself